homeassistant/components/filesize/sensor.py

```diff
@@ -52,48 +56,37 @@ def setup_platform(
 class Filesize(SensorEntity):
     """Encapsulates file size information."""
 
-    def __init__(self, path):
+    _attr_native_unit_of_measurement = DATA_MEGABYTES
+    _attr_icon = ICON
+
+    def __init__(self, path: str) -> None:
         """Initialize the data object."""
         self._path = path  # Need to check its a valid path
-        self._size = None
-        self._last_updated = None
-        self._name = path.split("/")[-1]
-        self._unit_of_measurement = DATA_MEGABYTES
+        self._size: int | None = None
+        self._last_updated: str | None = None
+        self._attr_name = path.split("/")[-1]
+        self._attr_unique_id = path
```

Comment:
This needs to be normalized and the schema needs to ensure that there are duplicates.

Reply:
I put in a check to ensure no duplicates and also a normalization so hope you find the suggestion fine.